src/Data/Dwarf/Reader.hs

```diff
@@ -23,14 +26,36 @@ module Data.Dwarf.Reader
   , drTarget64
   ) where
 
-import           Data.Binary.Get
-  ( Get
+import           Data.Binary.Get  ( Get
   , getWord16be, getWord32be, getWord64be
   , getWord16le, getWord32le, getWord64le
   )
+
+import           qualified Data.Binary.Get.Internal as GI
+
 import           Data.Int (Int16, Int32, Int64)
+import           qualified Data.ByteString as B
+import           qualified Data.ByteString.Unsafe as B
+import           qualified Data.Bits as Bits
+
 import           Data.Word (Word16, Word32, Word64)
 
+word24le :: B.ByteString -> Word32
+word24le s = fromIntegral (s `B.unsafeIndex` 0) Bits..|. 
+  (fromIntegral (s `B.unsafeIndex` 1) `Bits.unsafeShiftL` 8) Bits..|. 
+  (fromIntegral (s `B.unsafeIndex` 2) `Bits.unsafeShiftL` 16)
+
+word24be :: B.ByteString -> Word32
+word24be s = fromIntegral (s `B.unsafeIndex` 2) Bits..|.
+  (fromIntegral (s `B.unsafeIndex` 1) `Bits.unsafeShiftL` 1) Bits..|. 
+  (fromIntegral (s `B.unsafeIndex` 0) `Bits.unsafeShiftL` 16)
```

Comment:
Should we really be using all these "unsafe" operations? What are the conditions for using these functions safely? Are they guaranteed to be met? It seems like probably not because we're in a parsing context?

Reply:
these are safe in the context of `getWord24be` since the readN will read 3 bytes so 0-2 indeces are safe. This is the same as teh getWord32be + word32be functions ideally we would only expose the gets

Reply:
Agreed that this seems safe for `get*`. Perhaps we could also move these to `where`-clauses on the `get` functions to make it obvious that they are not (intended to be) used elsewhere, and to move the code that requires the precondition closer to the code that provides it? Probably also worth restating the above in a comment.